firmware/ventilator-controller-stm32/Core/Src/Pufferfish/Driver/BreathingCircuit/AlarmMuteService.cpp

```diff
@@ -24,19 +26,22 @@ void AlarmMuteService::transform(
   input_clock(current_time);
 
   if (alarm_mute.active) {
-    alarm_mute.remaining = (mute_max_duration - mute_duration_) / clock_scale;
+    int remaining =
```

Comment:
In general for the firmware, we should use `int32_t` or `int64_t` rather than `int`, just to make the width of the type explicit. It should be fine to declare `remaining` as `int64_t`, to keep the same width as the right-hand side of the assignment.

Reply:
I think I tried int64_t first but I guess util::clamp doesn't accept int64_t as a valid type?

Reply:
Here's the function declaration: https://github.com/pez-globo/pufferfish-software/blob/53f83d824b8db7964133b40d88b00ca4372238b9/firmware/ventilator-controller-stm32/Core/Inc/Pufferfish/Util/Ranges.h#L52

You may need to call it as `Util::clamp<int64_t>(...)`. I guess the template argument type deduction got confused if you gave it an `int64_t` and two integer literals `0` and `120` (which are probably `int32_t`s but I'm not sure); if you provided `mute_max_duration / clock_scale` (which evaluates to a `uint32_t`) instead of `120` and it wasn't working, the template argument type deduction would definitely fail because it expects all arguments to have the same type.

Reply:
Ya, I did pass integer literals, yep specifying the template type works, I'll test it and maybe you could check if it's fine now?

Reply:
seems fine on light testing. will merge once you resolve this conversation 👍🏽